clang/lib/Sema/Sema.cpp

```diff
@@ -995,12 +995,14 @@ void Sema::ActOnEndOfTranslationUnitFragment(TUFragmentKind Kind) {
     PerformPendingInstantiations();
   }
 
-  if (getLangOpts().SYCLIsDevice) {
+  if (getLangOpts().SYCLIsDevice)
     // Emit SYCL integration header for current translation unit if needed
     if (SyclIntHeader != nullptr)
       SyclIntHeader->emit(getLangOpts().SYCLIntHeader);
+
+  if (getLangOpts().SYCLIsDevice || getLangOpts().SYCLIsHost)
```

Comment:
Was it made to promote an attribute up to the kernel?

Reply:
@Fznamznon, I believe so.  I think @smanna12 is discussing with @turinevgeny, if the attribute still needs to be propagated up.

Reply:
Yes, this is under discussion. Will wait for @turinevgeny reply here.

Reply:
@turinevgeny do we need to propagate up the attribute to the kernel?

Reply:
@smanna12 We're evaluating this on a couple of workloads. Please stay tuned.